system/security/securesocket/securesocket.cpp

```diff
@@ -774,27 +774,20 @@ void CSecureSocket::readTimeout(void* buf, size32_t min_size, size32_t max_size,
         }
 
         rc = SSL_read(m_ssl, (char*)buf + size_read, max_size - size_read);
-        if(rc > 0)
+        if (rc > 0)
         {
             size_read += rc;
         }
         else
         {
             int err = SSL_get_error(m_ssl, rc);
-            // Ignoring SSL_ERROR_SYSCALL because IE prompting user acceptance of the certificate 
-            // causes this error, but is harmless.
-            // Ignoring SSL_ERROR_SYSCALL also seems to ignore the timeout value being exceeded,
-            //    but for persistence at least, treating zero bytes read can be treated as a graceful completion of connection
-            if((err != SSL_ERROR_NONE) && (err != SSL_ERROR_SYSCALL))
-            {
-                if(m_loglevel >= SSLogMax)
-                {
-                    char errbuf[512];
-                    ERR_error_string_n(ERR_get_error(), errbuf, 512);
-                    DBGLOG("Warning: SSL_read error %d - %s", err, errbuf);
-                }
-            }
-            break;
+            char errbuf[512];
+            ERR_error_string_n(err, errbuf, 512);
```

Comment:
Why err and not ERR_get_error() here ?

Reply:
maybe also call SSL_shutdown() ?

Reply:
ERR_get_error() returns the same error code, with the effect of taking the error off the stack. ERR_clear_error() does the same and will clear up everything.

Reply:
Don't believe we should call SSL_shutdown() here. Manual explicitly says not to call for certain errors.